src/buckets/components/CreateBucketForm.tsx

```diff
@@ -0,0 +1,108 @@
+// Libraries
+import React, {FC, ChangeEvent, FormEvent, useReducer} from 'react'
+import {useSelector, useDispatch} from 'react-redux'
+
+// Components
+import BucketOverlayForm from 'src/buckets/components/BucketOverlayForm'
+
+// Actions
+import {createBucketAndUpdate} from 'src/buckets/actions/thunks'
+
+// Constants
+import {CLOUD} from 'src/shared/constants'
+
+// Types
+import {
+  createBucketReducer,
+  RuleType,
+  initialBucketState,
+  DEFAULT_RULES,
+} from 'src/buckets/reducers/createBucket'
+import {AppState, Bucket} from 'src/types'
+
+// Selectors
+import {getOrg} from 'src/organizations/selectors'
+import {getBucketRetentionLimit} from 'src/cloud/utils/limits'
+import {getOverlayParams} from 'src/overlays/selectors'
+
+interface CreateBucketFormProps {
+  onClose: () => void
+}
+
+const CreateBucketForm: FC<CreateBucketFormProps> = props => {
+  const {onClose} = props
+  const org = useSelector(getOrg)
+  const isRetentionLimitEnforced = useSelector((state: AppState): boolean => {
+    if (CLOUD) {
+      return getBucketRetentionLimit(state)
+    }
+    return false
+  })
+  const overlayParams = useSelector(getOverlayParams)
+  const reduxDispatch = useDispatch()
+  const [state, dispatch] = useReducer(
+    createBucketReducer,
+    initialBucketState(isRetentionLimitEnforced, org.id)
+  )
+
+  const retentionRule = state.retentionRules.find(r => r.type === 'expire')
```

Comment:
```suggestion
  const retentionRule = state.retentionRules.find(rule => rule.type === 'expire')
```

Reply:
Oops, yeah. I actually had this changed in the other branch. When i created a new branch to get just this change, i forgot about this change. Good call. 👍